File: medicine_reminder_app/lib/models/adherence_log.dart
class AdherenceLog {
  final String id;
  final String userId;
  final String medicationId;
  final DateTime scheduledTime;
  final DateTime? takenTime;
  final String status; // 'taken', 'missed', 'skipped'

  AdherenceLog({
    required this.id,
    required this.userId,
    required this.medicationId,
    required this.scheduledTime,
    this.takenTime,
    required this.status,
  });

  factory AdherenceLog.fromJson(Map<String, dynamic> json) {
    return AdherenceLog(
      id: json['_id'],
      userId: json['userId'],
      medicationId: json['medicationId'],
      scheduledTime: DateTime.parse(json['scheduledTime']),
      takenTime: json['takenTime'] != null 
        ? DateTime.parse(json['takenTime']) 
        : null,
      status: json['status'],
    );
  }
}

File: backend/models/AdherenceLog.js
const mongoose = require('mongoose');

const adherenceLogSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  medicationId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Medication',
    required: true
  },
  scheduledTime: {
    type: Date,
    required: true
  },
  takenTime: Date,
  status: {
    type: String,
    enum: ['taken', 'missed', 'skipped'],
    required: true
  },
  notes: String,
  createdAt: {
    type: Date,
    default: Date.now
  }
});

module.exports = mongoose.model('AdherenceLog', adherenceLogSchema);